vars/buildDockerAndPublishImage.groovy

```diff
@@ -33,18 +37,44 @@ def call(String imageShortName, Map userConfig=[:]) {
   DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX")
   final String buildDate = dateFormat.format(now)
 
+  // only one platform parameter is supported for now either platform or platforms
+  if (finalConfig.platforms != '' && finalConfig.platform != '') {
+    // both platform and platforms cannot be set at the same time
+    echo 'ERROR: Only one platform parameter is supported for now either platform or platforms, prefer platforms.'
+    currentBuild.result = 'FAILURE'
+    return
+  } else if (finalConfig.platform != '') {
+    // if platform is set, I override platforms with it
+    finalConfig.platforms = finalConfig.platform
+  } else if (finalConfig.platforms == '') {
+    // if platforms is not set, I set it to linux/amd64 by default
+    finalConfig.platforms = 'linux/amd64'
+  }
```

Comment:
is there a reason to not add an `else { }` ? The condition set is hard to read if there is not clear happy path or default path

Reply:
simplified